admin/tabs/roles/editor/RoleEditorModel.ts

```diff
@@ -0,0 +1,109 @@
+import {RoleFormModel} from '@xh/hoist/admin/tabs/roles/editor/form/RoleFormModel';
+import {HoistRole} from '@xh/hoist/admin/tabs/roles/HoistRole';
+import {RolesModel} from '@xh/hoist/admin/tabs/roles/RolesModel';
+import {HoistModel, managed, TaskObserver, XH} from '@xh/hoist/core';
+import {Icon} from '@xh/hoist/icon';
+import {makeObservable} from '@xh/hoist/mobx';
+import {omit} from 'lodash';
+import {action, computed, observable} from 'mobx';
+
+export class RoleEditorModel extends HoistModel {
+    readonly rolesModel: RolesModel;
+    readonly savingTask = TaskObserver.trackLast({message: 'Saving Role'});
+    readonly deletingTask = TaskObserver.trackLast({message: 'Deleting Role'});
+
+    @managed readonly roleFormModel = new RoleFormModel();
+
+    @observable isOpen = false;
+    @observable role?: HoistRole;
+
+    private resolve: (role?: HoistRole) => void;
```

Comment:
I wasn't clear on how this is used

Reply:
It should've been called on save or cancel, but realize I missed calling it on cancel